examples/ConductorSharp.ApiEnabled/Services/TaskExecutionCounterService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using ConductorSharp.Engine.Interface;
+using System.Collections.Concurrent;
+
+namespace ConductorSharp.ApiEnabled.Services
+{
+    public class TaskExecutionCounterService : ITaskExecutionCounterService
+    {
+        private const int _maxTrackedCount = 10000;
+
+        private readonly ConcurrentDictionary<string, RunningTask> _inProgress = new();
+        private readonly ConcurrentDictionary<string, int> _failed = new();
+        private readonly ConcurrentDictionary<string, int> _completed = new();
+
+        public Task OnPolled(RunningTask trackedTask)
+        {
+            if (_inProgress.Keys.Count < _maxTrackedCount)
+            {
+                _inProgress.TryAdd(trackedTask.TaskId, trackedTask);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public Task OnFailed(RunningTask trackedTask)
+        {
+            if (_inProgress.TryGetValue(trackedTask.TaskId, out var task))
+            {
+                _inProgress.Remove(task.TaskId, out var removedTask);
+                _failed.AddOrUpdate(task.TaskName, 1, (id, count) => count + 1);
```

Comment:
Delegates are executed outside the locks, hence this operation is not atomic. Check the [docs](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-7.0)

Reply:
Did not realize this, thanks. I also removed the size limitation on this dictionary as this is just an example project.